From 6aed1fa39f21b7db8ab28e52e7dafdf6a411fccb Mon Sep 17 00:00:00 2001
From: Harshitha Prem <quic_hprem@quicinc.com>
Date: Thu, 1 Dec 2022 23:24:35 +0530
Subject: [PATCH] wpa_supplicant: mesh 6G incorrect channel bandwidth

when a phy supports more than one 80211 modes, for
6g mesh interface it is not able to map the correct
channel width because of the mode of 5G and 6G interface
are same in supported modes

Hence, get the phy mode based on the channel freq.

Signed-off-by: Harshitha Prem <quic_hprem@quicinc.com>
---
 wpa_supplicant/wpa_supplicant.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/wpa_supplicant/wpa_supplicant.c b/wpa_supplicant/wpa_supplicant.c
index 0ee9be5..7b9ba42 100644
--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -2718,13 +2718,8 @@ void ibss_mesh_setup_freq(struct wpa_supplicant *wpa_s,
 	}
 
 	hw_mode = ieee80211_freq_to_chan(freq->freq, &channel);
-	for (i = 0; wpa_s->hw.modes && i < wpa_s->hw.num_modes; i++) {
-		if (wpa_s->hw.modes[i].mode == hw_mode) {
-			mode = &wpa_s->hw.modes[i];
-			break;
-		}
-	}
-
+	mode = get_mode(wpa_s->hw.modes, wpa_s->hw.num_modes,
+			hw_mode, is_6ghz_freq(ssid->frequency));
 	if (!mode)
 		return;
 
-- 
2.17.1

